ethereum/test/profileTest.js

```diff
@@ -0,0 +1,104 @@
+const fs = require('fs')
+const BNBToken = artifacts.require("BNBToken");
+const ERC20AtomicSwapper = artifacts.require("ERC20AtomicSwapper");
+const truffleAssert = require('truffle-assertions');
+const calculateSecretHashLock = require('./secretHashLock')
+
+let profile;
+try {
+    profile = require('../GasProfile.json') || {};
+} catch (error) {
+    if (!error.message.startsWith('Cannot find module')) {
+        console.error(error);
+    }
+    console.log('Creating GasProfile.json')
+    profile = {}
+}
+
+function showRegression(type, actual) {
+    const expected = profile[type];
+    if (actual < expected) {
+        console.log("\x1b[32m", type, "improvement:", expected, '->', actual, "\x1b[0m");
+    } else if (actual > expected) {
+        console.log("\x1b[31m", type, "regression:", expected, '->', actual, "\x1b[0m");
+    } else if (typeof(expected) === 'undefined') {
+        console.log(type, '=', actual)
+    }
+    profile[type] = actual
+}
+
+function showRegressions(results) {
+    for (let trial in results) {
+        showRegression(trial, results[trial]);
+    }
+}
+
+
+function hasNoZero(address) {
+    for (let i = 2; i < address.length; i++) {
+        if (address.substr(i, 2) == '00') {
+            return false
+        }
+    }
+    return true
+}
+
+
+contract('ERC20AtomicSwapper', (accounts) => {
+    const [_, owner, operator, swapA, swapB] = accounts.filter(hasNoZero)
+    // Constant swap parameters
+    const timestamp = 1565312187607;
```

Comment:
```
const timestamp = Math.floor(Date.now()/1000);
```
Thus you don't need to remove the timestamp constrain in contract.

Reply:
Not true. If any of the timestamp bytes' zeroeness change, the measured gas will differ. This is because zero-bytes in message data cost 4 gas, while non-zero bytes cost 68 gas.

Reply:
https://ethereum.github.io/yellowpaper/paper.pdf